smartling-job-batches-api/src/main/java/com/smartling/api/jobbatches/util/FileUploadProxyUtils.java

```diff
@@ -0,0 +1,150 @@
+package com.smartling.api.jobbatches.util;
+
+import com.smartling.api.v2.client.exception.RestApiExceptionHandler;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.jboss.resteasy.annotations.providers.multipart.PartFilename;
+import org.jboss.resteasy.annotations.providers.multipart.PartType;
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.jboss.resteasy.client.jaxrs.internal.ClientResponse;
+import org.jboss.resteasy.plugins.providers.multipart.FieldEnablerPrivilegedAction;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataOutput;
+
+import javax.ws.rs.FormParam;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.util.Map;
+
+@Slf4j
+public class FileUploadProxyUtils
```

Comment:
File upload proxying needed because neither Jax-Rs nor RestEasy doesn't provide a possibility to pass a dynamic number of parameters with dynamically given names (file directives).
I temporarily moved duplicated code to this utils
Our next planned SDK is FilesAPI, and it will require the same hacks, and we are going to heavily refactor this and move to commons

Reply:
👍